### ISSUE TITLE
[new-backend] spec: reimplement to completely comply with hooks

### ISSUE THREAD

**Issue by atmaxinger** [ElektraInitiative/libelektra#4500]
The spec plugin isn't completely conformant with what should be done in the spec hook. When implementig the hook, we decided to not waste too much time on it, just that the basisc work.

---

> There can be different settings for the conflict handling
> * This can probably be removed

Yes, all the conflict handling settings should be removed entirely. During `get` all conflicts should be warnings during `set` they should be errors (unless @markus2330 thinks some conflicts should never be errors). That part could be handled from the `libelektra-kdb` side with the "error blocking trick", so the `spec` would be the same.

> neither `assign/condition` nor `default` are processed in set

Not sure about `assign/condition` and whether we actually want to keep that (@markus2330 ?), but `default` should definitely be processed in `set`. See also https://github.com/ElektraInitiative/libelektra/pull/4484#discussion_r979454050

> * set operation sets `internal/spec/remove` on `array`
> * set removes `internal/spec/array/validated`
> * set does this thing here, I'm not sure what it is

All of the stuff related to array handling is very complicated and weird. Part of the weirdness comes from the fact that on `master` the `spec` plugin is only called once during `kdbSet` and somehow there should still be validation during `kdbSet`, but also `spec` should cleanup the copied data (obviously not possible).

See also #2700 and #4061

---

However, please don't waste time on this. Like I said already: Just try to imitate what `spec` does on `master`. The existing tests obviously also expect this behaviour. Once that is done and merged, you can create another PR with the actual updates for `spec` (probably after `new-backend` is merged in `master`).

_Originally posted by @kodebach in https://github.com/ElektraInitiative/libelektra/pull/4495#discussion_r982213541_

**Comment by kodebach** [comment]
Another important thing is that `default:/` keys should be processed in the `poststorage` phase of `kdbGet`, currently the are temporarily cut out during that phase.

**Comment by atmaxinger** [comment]
Yes - See also #4493

**Comment by markus2330** [comment]
@atmaxinger is this done?

**Comment by atmaxinger** [comment]
Well as @tmakar has rewritten the spec plugin, I think this is done.
